lib/api.rb

```diff
@@ -243,59 +314,159 @@ def validation_error(msg)
     }.to_json
   end
 
-  #
-  # ciphers
-  #
-
-  # create a new cipher
-  post "/ciphers" do
+  # Used by the web vault to update the private and public keys if the user doesn't have one.
+  post "/accounts/keys" do
+    content_type :json
+    # Needed by the web vault for EVERY response
+    response['access-control-allow-origin'] = '*'
     d = device_from_bearer
     if !d
       return validation_error("invalid bearer")
     end
 
-    need_params(:type, :name) do |p|
-      return validation_error("#{p} cannot be blank")
+    if !params[:encryptedprivatekey].to_s.match(/^2\..+\|.+/)
+      return validation_error("Invalid key")
     end
 
-    begin
-      Bitwarden::CipherString.parse(params[:name])
-    rescue Bitwarden::InvalidCipherString
-      return validation_error("Invalid name")
+    d.user.private_key = params[:encryptedprivatekey]
+    d.user.public_key = params[:publickey]
+
```

Comment:
Don't you need to save the user record to retain the private key and public key?

Reply:
It's done like the others parameters just above, and checked with the if statement line 288.

Reply:
do we need to save the private and public key to the database?